frontend/device/kobo/device.lua

```diff
@@ -639,9 +718,11 @@ function Kobo:init()
             mxcfb_bypass_wait_for = mxcfb_bypass_wait_for,
         }
         if self.screen.fb_bpp == 32 then
-            -- Ensure we decode images properly, as our framebuffer is BGRA...
-            logger.info("Enabling Kobo @ 32bpp BGR tweaks")
-            self.hasBGRFrameBuffer = yes
+            if not self.isRGB then
+                -- Ensure we decode images properly, as our framebuffer is BGRA...
+                logger.info("Enabling Kobo @ 32bpp BGR tweaks")
+                self.hasBGRFrameBuffer = yes
+            end
```

Comment:
Pinging @NiLuJe on this one in case there's a better way.

Besides that, in the meantime you could write this like `if self.screen.fb_bpp == 32 and not self.isRGB then`

Reply:
I'd probably rename `isRGB` to `hasColorPanel`, but otherwise I don't have anything much to say about the logic; as we do need to not break the never-used-on-purpose-but-actually-working 32bpp codepaths on *all the other* Kobo devices :).

I wholeheartedly agree with your layout proposition, though.

Reply:
Oh, and to expand on your original comment about function vs. boolean and/or private vs. public, it's *probably* something we'd want to be public, juuust in case.

Reply:
Which, obviously, made me realize that, duh', we already do have an existing device cap for exactly that: `hasColorScreen` ^^

Reply:
TL;DR: Switch to `hasColorScreen = no,` and `if self.screen.fb_bpp == 32 and not self:hasColorScreen() then`

Reply:
Oh, and you're already setting `hasColorScreen` anyway, good ;).